src/compat.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2019-2020 Uwe Bonnes <bon@elektron.ikp.physik.tu-darmstadt.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.	 If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* Compatibility functions wrapping compiler-specific code, especially
+ * built-ins. */
+
+#include "compat.h"
+
+#ifdef __GNUC__
+#define unlikely(x) __builtin_expect(x, 0)
+#else
+#define unlikely(x) x
+#endif
+
```

Comment:
The extra blank line here introduces a whitespace error, please remove it

Reply:
Done. Personally I prefer some visual separation between defines and functions but that's a matter of taste.

Reply:
The code base style does not include this extra space and uses a single blank line per the clang-format configuration - so it's more what we enforce across the whole code base for consistency rather than taste. The project has had a history of people cargo-culting their personal style in which caused a huge number of headaches in the lead-up to v1.8.